tests/python/test_python_frontend.py

```diff
@@ -4584,6 +4584,55 @@ def fusion_func(
                     )
                 torch.testing.assert_close(nvf_out[0], ref_out)
 
+    # Test that replaced sizes using input tensor metadata are successfully computed
+    # See https://github.com/NVIDIA/Fuser/pull/2714 which surfaced this in
+    # failing thunder test
+    # thunder.tests.test_core.test_bsym_toposort_nvfuser_cuda_thunder.dtypes.float32
+    def test_replaced_sizes_pr2714(self):
```

Comment:
This test fails currently. We wind up needing to compute metadata for T0 at runtime, but we can't because we never actually _bind_ `GetMetadata` to `PrecomputedValues`.  It looks at a glance like this line is enough:https://github.com/NVIDIA/Fuser/blob/862dfd18a03b48617c4789b7507455eb996fb5d3/csrc/evaluator_common.cpp#L373
However, `bindValue` requires `evaluatorIndex()` to be set on the `Val` for it to have any effect: https://github.com/NVIDIA/Fuser/blob/862dfd18a03b48617c4789b7507455eb996fb5d3/csrc/evaluator_common.h#L208-L211. The only `Val`s that have evaluator indices are those found in `collectRuntimeUsedValues`: https://github.com/NVIDIA/Fuser/blob/862dfd18a03b48617c4789b7507455eb996fb5d3/csrc/evaluator_common.cpp#L93
This function doesn't currently look for `metadataOf(tv)` and simply adding that line doesn't seem to be enough. I will investigate this more; it seems like a more general bug but I haven't yet isolated a simpler repro for just this particular failure.

Reply:
With latest commit c423854 this gets closer, but still fails in `PrecomputedValues::validate` at this check: https://github.com/NVIDIA/Fuser/blob/5326aff33d3078f03a103d2486a672ba3b6ea1d6/csrc/evaluator_common.cpp#L317
The `TensorMetaData` is held in a `PolymorphicValue` that is a `StructHandle` and `isSame` just checks that the pointer is equal. https://github.com/NVIDIA/Fuser/blob/5326aff33d3078f03a103d2486a672ba3b6ea1d6/csrc/polymorphic_value.h#L279 We might be creating metadata multiple times for some reason. I am not sure. However, if I just insert
```c++
if (a.is<StructHandle>()) {
  return b.is<StructHandle>();
}
```
then this test passes. That weakens the validation of course. A proper fix would do a deep equality check, probably by special casing for known `Struct`s like `TensorMetaData`. For now, maybe we can avoid this pointer check for `Struct`s? What do you think, @zasdfgbnm ?

Reply:
Current status: the deep validation check was merged (#2811), but there are still some CI errors blocking #2812 which actually binds the tensor metadata. Once that is figured out this PR should be ready to go.

Reply:
This now passes.